src/store/keyboard/sagas.ts

```diff
@@ -1,11 +1,30 @@
-import { put } from 'redux-saga/effects'
+import { fork, put, take } from 'redux-saga/effects'
 import { Keyboard } from 'react-native'
 
 import { open, close } from './slice'
+import { eventChannel } from 'redux-saga'
 
-export function* initKeyboardEvents() {
-  Keyboard.addListener('keyboardDidShow', () => put(open))
-  Keyboard.addListener('keyboardDidHide', () => put(close))
+function* initKeyboardEvents() {
+  const keyboardChannel = eventChannel(emitter => {
+    Keyboard.addListener('keyboardDidShow', () => {
+      emitter('show')
+    })
+    Keyboard.addListener('keyboardDidHide', () => {
+      emitter('hide')
+    })
+    return () => {}
+  })
+
+  yield fork(function* () {
```

Comment:
what does the fork gain here? probably can just put the while (true) { } in line right?